src/box/msgpack.c

```diff
@@ -84,9 +84,30 @@ msgpack_snprint_ext(char *buf, int size, const char **data, int depth)
 	}
 }
 
+/** Our handler to validate MP_EXT contents. */
+static int
+msgpack_check_ext(const char *data, int8_t type, uint32_t len)
+{
+	switch (type) {
+	case MP_DECIMAL:
+		return mp_check_decimal(data, len);
+	case MP_UUID:
+		return mp_check_uuid(data, len);
+	case MP_DATETIME:
+		return mp_check_datetime(data, len);
```

Comment:
These functions look confusing: we have `mp_encode_datetime` and `mp_decode_datetime`, which behave exactly like `mp_encode_uint` and `mp_decode_uint`, i.e. they produce a valid msgpack, but at the same time `mp_check_datetime` is completely different from `mp_check_uint`. IMO better not introduce these helpers at all and use `decimal_unpack` directly here; or you should name these helpers so that they don't look like `mp_check_uint` and friends.

Reply:
What do you think of `mp_validate_<type>` ? 
I renamed all the functions to this template in v2.